### PR TITLE
 Check for main waypoint before deleting association

### LINKED ISSUE #273: Check for main waypoint before deleting association
Adapt the association web-service so that an association between waypoint and route can not be deleted if the waypoint is main waypoint of the route.

See https://github.com/c2corg/v6_ui/issues/298#issuecomment-218394059

### PR DESCRIPTION
**Pull request by tsauerwein**
Closes #273

### CODE REVIEW

File: c2corg_api/views/association.py [hunk 1]
```diff
@@ -76,6 +78,9 @@ def collection_delete(self):
             if association is None:
                 raise HTTPBadRequest('association does not exist')
 
+        if is_main_waypoint_association(association):
+            raise HTTPBadRequest('waypoint is still main waypoint of route')
```
Comment (asaunier) [inline c2corg_api/views/association.py]: Perhaps the error could be more explicit, like ``As the main waypoint of the route, this waypoint cannot be deassociated``?

**Comment by asaunier** [comment]
Only a minor comment, else looks good to me, thanks.

File: c2corg_api/views/association.py (hunk 1 above)
Comment (tsauerwein) [inline c2corg_api/views/association.py]: Yes, that makes it a bit clearer. But remember that the error messages that the API returns are not intended to be shown directly to the user.

Comment (asaunier) [inline c2corg_api/views/association.py]: We never know... :P

**Pull request merged**